core/src/main/java/org/apache/carbondata/core/scan/filter/partition/PartitionFilterUtil.java

```diff
@@ -63,6 +64,9 @@ public static BitSet getPartitionMapForRangeFilter(PartitionInfo partitionInfo,
           for (String value : values.get(i)) {
             Object listValue = PartitionUtil.getDataBasedOnDataType(value, partitionColumnDataType,
                 timestampFormatter, dateFormatter);
+            if (listValue instanceof String) {
+              listValue = ByteUtil.toBytes((String)listValue);
+            }
```

Comment:
please improve performance and refractory code.
we should convert value only once. please move the code to the outside of for loop.